src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpResponseMessageHelper.cs

```diff
@@ -173,16 +173,16 @@ private Task<Message> ReadStreamAsMessageAsync()
             }
             if (!content.Headers.ContentLength.HasValue)
             {
-                return ReadChunkedBufferedMessageAsync(contentStreamTask);
+                return ReadChunkedBufferedMessageAsync(contentStreamTask, timeoutHelper);
             }
             return ReadBufferedMessageAsync(contentStreamTask);
         }
 
-        private async Task<Message> ReadChunkedBufferedMessageAsync(Task<Stream> inputStreamTask)
+        private async Task<Message> ReadChunkedBufferedMessageAsync(Task<Stream> inputStreamTask, TimeoutHelper timeoutHelper)
         {
             try
             {
-                return await _encoder.ReadMessageAsync(await inputStreamTask, _factory.BufferManager, _factory.MaxBufferSize, _contentType);
+                return await _encoder.ReadMessageAsync(await inputStreamTask, _factory.BufferManager, _factory.MaxBufferSize, _contentType, await timeoutHelper.GetCancellationTokenAsync());
```

Comment:
I originally made this change as I was trying to get the cancellation token passed through to enable pre-reading the stream asynchronously with the timeout applied. Unfortunately this couldn't be done in the end due to needing to cross over a synchronous public api in MessageEncoder to be able to put the ReadAheadWrappingStream in the correct place for the call hierarchy to work.